src/Dictionaries.jl

```diff
@@ -13,6 +13,20 @@ export dictionary, index, distinct, disjoint, isdictequal, filterview, sortkeys,
 export issettable, isinsertable, set!, unset!
 export istokenizable, tokentype, tokens, tokenized, gettoken, gettokenvalue, istokenassigned, settokenvalue!, gettoken!, deletetoken!, sharetokens
 
+"""
+    return_type(f, types)
+
+Find the return type of `f` called with `types`.
+
+!!! note 
+    Currently, this method depends on `code_typed`.
+"""
+function return_type(f, types)
+    return last(only(code_typed(f, types; 
```

Comment:
This is worse than just calling `Core.Compiler.return_type`. It still trespasses on the internals of the Julia implementation, as it's not documented that `last` may be used here, and I guess it's slower than `Core.Compiler.return_type`.

Reply:
@nsajko It's calling `last` on a `Pair` -- how is that undocumented behavior?  It's also only calling public API (an function that appears in the manual). I guess I could also access the `rettype` property of the `CodeInfo` that's the first element of that pair, but I still need to access the pair.

It will almost definitely be slower than calling `Core.Compiler.return_type`, because I'm guessing that `Core.Compiler.return_type` is called somewhere in the implementation of `code_typed`.

Reply:
You're right, sorry.